### PR TITLE
HAAR-30: view all prisons screen

### PR DESCRIPTION
**Pull request by neilmendum**
https://dsdmoj.atlassian.net/browse/HAAR-30

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -21,9 +21,9 @@ services:
     healthcheck:
       test: ["CMD", "curl", "-f", "http://localhost:8080/auth/health"]
     environment:
-      SPRING_PROFILES_ACTIVE: dev,token-verification
+      SPRING_PROFILES_ACTIVE: dev,token-verification,nomis
```
Comment (neilmendum) [inline docker-compose.yml:24]: This lets us login locally with NOMIS accounts

File: docker-compose.yml [hunk 2]
```diff
@@ -74,19 +78,50 @@ services:
...
+
+  prison-register:
+    image: quay.io/hmpps/prison-register:latest
+    networks:
+      - hmpps
+    container_name: prison-register
+    depends_on:
+      - hmpps-auth
+    ports:
+      - "8103:8080"
```
Comment (neilmendum) [inline docker-compose.yml:91]: Used same port mapping as https://github.com/ministryofjustice/send-legal-mail-to-prisons/blob/main/docker-compose.yml#L49

File: server/routes/index.ts [hunk 3]
```diff
@@ -24,6 +27,14 @@ export default function routes(router: Router, services: Services): Router {
           roles: [MAINTAINER_ROLE],
           enabled: true,
         },
+        {
+          id: 'prison-register',
+          heading: 'Prison register',
+          description: 'View prison details.',
+          href: '/prison-register',
+          roles: [MAINTAINER_ROLE],
```
Comment (neilmendum) [inline server/routes/index.ts:35]: Using the same role as Court Register tile as haven't heard it should be a different one

File: docker-compose.yml [hunk 4]
```diff
@@ -74,19 +78,50 @@ services:
...
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://localhost:8080/health/ping"]
+    environment:
+      - SERVER_PORT=8080
+      - SPRING_PROFILES_ACTIVE=dev
```
Comment (neilmendum) [inline docker-compose.yml:96]: Using `dev` profile which spins up in memory H2 db, it didn't seem necessary to instantiate another Postgres instance like Court Register uses

File: docker-compose.yml [hunk 5]
```diff
@@ -57,14 +59,16 @@ services:
     environment:
       - SERVER_PORT=8080
       - OAUTH_ENDPOINT_URL=http://hmpps-auth:8080/auth
-      - SNS_ENDPOINT_URL=http://localstack:4575
-      - SPRING_PROFILES_ACTIVE=stdout,localstack
+      - SPRING_DATASOURCE_URL=jdbc:postgresql://court-register-db:5432/court_register_db
+      - SNS_ENDPOINT_URL=http://localstack:4566
+      - HMPPS_SQS_LOCALSTACKURL=http://localstack:4566
```
Comment (Mjwillis) [inline docker-compose.yml:64]: Why does front end need localstack?

File: docker-compose.yml [hunk 6]
```diff
@@ -74,19 +78,50 @@ services:
...
+    ports:
+      - "8082:8080"
+    healthcheck:
+      test: [ "CMD", "curl", "-f", "http://localhost:8080/health" ]
+    environment:
+      - SERVER_PORT=8080
+      - SPRING_PROFILES_ACTIVE=dev
+      - API_BASE_URL_OAUTH=http://hmpps-auth:8080/auth
 
   localstack:
```
Comment (Mjwillis) [inline docker-compose.yml:115]: Can't remember why the UI needs localstack?

**Review by Mjwillis — COMMENTED** [review]
Just a question about localstack

File: docker-compose.yml (hunk 6 above)
Comment (neilmendum) [reply docker-compose.yml:115]: That's because the UI depends on `court-register` which needs localstack

File: server/@types/prisonRegisterImport/index.d.ts [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+    responses: {
+      /** Successful Operation */
+      200: {
+        content: {
+          'application/json': components['schemas']['PrisonDto']
+        }
+      }
+    }
+  }
+  getPrisonFromId_1: {
```
Comment (karenmillermoj) [inline server/@types/prisonRegisterImport/index.d.ts:276]: Is this getPrisonFromId_1 used?

Comment (neilmendum) [reply server/@types/prisonRegisterImport/index.d.ts:276]: Not currently. That file was generated when I ran `npx openapi-typescript` and I didn't make any changes to it

I think it appended `_1` because `prison-register` has two endpoints with the same path variable:
```
'/prisons/id/{prisonId}': {
  /** Information on a specific prison */
  get: operations['getPrisonFromId']
}
'/gp/prison/{prisonId}': {
  get: operations['getPrisonFromId_1']
}
```

File: docker-compose.yml [hunk 8]
```diff
@@ -74,19 +78,50 @@ services:
...
-    image: localstack/localstack:0.11.2
+    image: localstack/localstack:0.12.10
```
Comment (karenmillermoj) [inline docker-compose.yml:116]: Latest local stack is 0.14.0 - happy for this to go in another PR

**Review by Mjwillis — APPROVED** [review]
LGTM
